lib/datadog/appsec/event.rb

```diff
@@ -98,16 +92,15 @@ def self.record_via_span(*events)
           # apply tags to root span
 
           # complex types are unsupported, we need to serialize to a string
-          triggers = root_tags.delete('_dd.appsec.triggers')
-          root_span.set_tag('_dd.appsec.json', JSON.dump({ triggers: triggers }))
+          triggers = trace_tags.delete('_dd.appsec.triggers')
+          trace.send(:set_tag, '_dd.appsec.json', JSON.dump({ triggers: triggers }))
 
-          root_tags.each do |key, value|
-            root_span.set_tag(key, value.is_a?(String) ? value.encode('UTf-8') : value) unless root_span_tags.include?(key)
+          trace_tags.each do |key, value|
+            trace.send(:set_tag, key, value.is_a?(String) ? value.encode('UTF-8') : value)
```

Comment:
Just FYI I think `send` might not be performant. Once made public, we should use `set_tag` instead.

Reply:
Definitely, although I don't see that being much of a concern currently given the number of calls to `set_tag` here, for which the difference in using `send` is dwarfed by the remainder of what is done by AppSec.

Reply:
Addressed